packages/user-operation-controller/src/UserOperationController.ts

```diff
@@ -404,6 +510,36 @@ export class UserOperationController extends BaseController<
     this.#updateMetadata(metadata);
   }
 
+  async #approveUserOperation(
+    metadata: UserOperationMetadata,
+    request: AddUserOperationRequest,
+    options: AddUserOperationOptions,
+  ) {
+    log('Requesting approval');
+
+    const { resultCallbacks, value: rawValue } = await this.#requestApproval(
+      metadata,
+    );
+
+    const value = rawValue as { txMeta?: TransactionMeta } | undefined;
+    const updatedTransaction = value?.txMeta;
```

Comment:
Suggestion: move this rawValue cast into #requestApproval and change that method's return type.

Reply:
Done.